### PR TITLE
Plip 1483 retina image scales

### PR DESCRIPTION
**Pull request by didrix**
plip: https://github.com/plone/Products.CMFPlone/issues/1483
This merge depend on https://github.com/plone/Products.CMFPlone/pull/1611

### CODE REVIEW

File: plone/namedfile/scaling.py [hunk 1]
```diff
@@ -341,6 +360,15 @@ def getImageSize(self, fieldname=None):
     def guarded_orig_image(self, fieldname):
         return guarded_getattr(self.context, fieldname, None)
 
+    def getRetinaScales(self):
+        try:
+            from Products.CMFPlone.utils import getRetinaScales
+        except ImportError:
```
Comment (ebrehault) [inline plone/namedfile/scaling.py]: Your other PR (https://github.com/plone/Products.CMFPlone/pull/1611) does implement `getRetinaScales`, why do you want to try/except this import?

File: plone/namedfile/scaling.py [hunk 2]
```diff
@@ -387,6 +415,30 @@ def scale(
         )
         if info is None:
             return  # 404
+
+        try:
+            (orig_width, orig_height) = self.getImageSize(fieldname)
+        except:
```
Comment (ebrehault) [inline plone/namedfile/scaling.py]: In which cases will we get an exception here?

File: plone/namedfile/scaling.py (hunk 1 above)
Comment (mauritsvanrees) [inline plone/namedfile/scaling.py]: Trying the import makes it easier to support earlier Plone versions that do not have this functionality. At Zest Software we may want to backport some of this to Plone 4.

We slightly refactored this in a065e173ad8b725403e5ee3a31904ea47f614494 to try the import only once at the top of the `scaling.py` file.

File: plone/namedfile/scaling.py (hunk 2 above)
Comment (mauritsvanrees) [inline plone/namedfile/scaling.py]: You can get an Unauthorized error when getting the original image object.

We changed this a bit in a065e173ad8b725403e5ee3a31904ea47f614494 but that caused Unauthorized errors in tests (plone.app.caching, plone.formwidget.namedfile, plone.outputfilters). I can't reproduce this Unauthorized error in real life.

In 0f4f960f475f6cdbcc653799bfce9acb41e84ed5 I changed it to specifically catch the Unauthorized error.

**Pull request merged**